--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -6615,7 +6615,11 @@ static void igb_reuse_rx_page(struct igb
 
 static inline bool igb_page_is_reserved(struct page *page)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,6,0)
 	return (page_to_nid(page) != numa_mem_id()) || page->pfmemalloc;
+#else
+	return (page_to_nid(page) != numa_mem_id());
+#endif
 }
 
 static bool igb_can_reuse_rx_page(struct igb_rx_buffer *rx_buffer,
