--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -4984,7 +4984,10 @@ static void igb_tx_map(struct igb_ring *
 	/* Make sure there is space in the ring for the next send. */
 	igb_maybe_stop_tx(tx_ring, DESC_NEEDED);
 
-	if (netif_xmit_stopped(txring_txq(tx_ring)) || !skb->xmit_more) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+	if (netif_xmit_stopped(txring_txq(tx_ring)) || !skb->xmit_more)
+#endif
+	{
 		writel(i, tx_ring->tail);
 
 		/* we need this if more than one processor can write to our tail
