--- a/net/nfc/netlink.c
+++ b/net/nfc/netlink.c
@@ -28,7 +28,7 @@
 #include "nfc.h"
 #include "llcp.h"
 
-static const struct genl_multicast_group nfc_genl_mcgrps[] = {
+static __genl_const struct genl_multicast_group nfc_genl_mcgrps[] = {
 	{ .name = NFC_GENL_MCAST_EVENT_NAME, },
 };
 
@@ -1489,7 +1489,7 @@ static int nfc_genl_se_io(struct sk_buff
 	return nfc_se_io(dev, se_idx, apdu, apdu_len, se_io_cb, ctx);
 }
 
-static const struct genl_ops nfc_genl_ops[] = {
+static __genl_const struct genl_ops nfc_genl_ops[] = {
 	{
 		.cmd = NFC_CMD_GET_DEVICE,
 		.doit = nfc_genl_get_device,
