--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -196,7 +196,9 @@ static const struct dev_pm_ops igb_pm_op
 };
 #endif
 static void igb_shutdown(struct pci_dev *);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
 static int igb_pci_sriov_configure(struct pci_dev *dev, int num_vfs);
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0) */
 #ifdef CONFIG_IGB_DCA
 static int igb_notify_dca(struct notifier_block *, unsigned long, void *);
 static struct notifier_block dca_notifier = {
@@ -237,7 +239,9 @@ static struct pci_driver igb_driver = {
 	.driver.pm = &igb_pm_ops,
 #endif
 	.shutdown = igb_shutdown,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
 	.sriov_configure = igb_pci_sriov_configure,
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0) */
 	.err_handler = &igb_err_handler
 };
 
@@ -7567,6 +7571,7 @@ static int igb_sriov_reinit(struct pci_d
 	return 0;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
 static int igb_pci_disable_sriov(struct pci_dev *dev)
 {
 	int err = igb_disable_sriov(dev);
@@ -7576,6 +7581,7 @@ static int igb_pci_disable_sriov(struct
 
 	return err;
 }
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0) */
 
 static int igb_pci_enable_sriov(struct pci_dev *dev, int num_vfs)
 {
@@ -7593,6 +7599,7 @@ out:
 }
 
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
 static int igb_pci_sriov_configure(struct pci_dev *dev, int num_vfs)
 {
 #ifdef CONFIG_PCI_IOV
@@ -7603,6 +7610,7 @@ static int igb_pci_sriov_configure(struc
 #endif
 	return 0;
 }
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0) */
 
 #ifdef CONFIG_NET_POLL_CONTROLLER
 /* Polling 'interrupt' - used by things like netconsole to send skbs
